### PR TITLE
Merging mkcloudruns with automation repository.

### PR DESCRIPTION
**Pull request by dguitarbite**
Mkcloudruns repository is a good starting point for the team to
collaborate on unifying all the commonly used configuration options,
variables and more for development and testing environments. This should
address new comers and veterans equally by helping them save a lot of
time. Hopefully we agree on a common policy to update the wrapper
scripts here with new handy variables and options that are eventually
added to the automation repository.

The very need for something like this is a good indicator that
automation repository is having some inherent flaws in the way it is
designed and is being updated. The final aim should be to try and fix
automation repository in the right way so that we do not need to use
these wrapper scripts to eliminate the boilerplate for deploying various
flavours of SUSE OpenStack Cloud.

Note: These scripts are being massively updated, modified and the
experience I have gathered from trainings, personal deployments for
testing and development of various features for Crowbar and OpenStack
are visible in form of code added under the mkcloudruns folder below.

Initial repository: https://github.com/dguitarbite/mkcloudruns

I may be updating some things in this repository over time but should
not be relied upon anymore. Rather clone automation repository and
modify the scripts in there.

### CODE REVIEW

**Comment by nicolasbock** [comment]
@dguitarbite I haven't had time to fully digest all of this, but let me give you big Thanks! for putting it here already! That's awesome!

File: mkcloudruns/README.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+This repository is basically a wrapper which automates the boilerplate required
+to start mkcloud with the right configuration settings. Even for non-beginners
+this boilerplate wrapper script should be really handy due to the overwhelming
+amount of configuration options and environment variables used by mkcloud :|.
+
+- Inspiration:
+    > I'm just too lazy.
+
+<<<<<<< HEAD
```
Comment (aspiers) [inline mkcloudruns/README.md]: Looks like an unresolved merge conflict here ...

File: mkcloudruns/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+>>>>>>> 6acf6db... Stub temp.
+
+# Deploying SUSE CLoud
+
+Follow these steps to deploy the required SUSE Cloud setup.
+
+## Initial Setup
+
+* Clone the repository
+  * These wrappers would eventually be part of automation repository, so I am trying to serve both worlds here ... use your common sense ;).
```
Comment (aspiers) [inline mkcloudruns/README.md]: You can remove this from the PR :-)

File: mkcloudruns/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+* Setup up libvirt, KVM,LVM as per the automation repo, follow (this link)[https://github.com/dguitarbite/automation/blob/master/docs/mkcloud.md]
```
Comment (aspiers) [inline mkcloudruns/README.md]: Wrong link syntax here. I suggest viewing the rendered HTML version at https://github.com/dguitarbite/automation/tree/master/mkcloudruns to check for other errors.

**Comment by aspiers** [comment]
I ran out of review time and going on FTO tomorrow, but I'll definitely check this out when I get back. Thanks a lot for sharing!

**Comment by dguitarbite** [comment]
@aspiers Thanks for reviewing till here. I would have to update the code a lot more I guess. Feel free to comment as an when time permits, I am not willing to let this get merged without at least 10+ +1's and no -1's. Another point is how to share the credit, some of the code is inspired from others work and some is probably other's work in form of gists, snippets here and there like the Support Enablement training early this year, how do I do this? Co authored??? Welcome, but I would be cheating if I took at the credit for this.

**Comment by dirkmueller** [comment]
whats the point of nodeXML file?

File: mkcloudruns/hacks/inject_stuff [hunk 4]
```diff
@@ -0,0 +1,4 @@
+for i in {1..15};
```
Comment (dirkmueller) [inline mkcloudruns/hacks/inject_stuff]: hardcoded ?

File: mkcloudruns/hacks/conduitresolver.rb [hunk 5]
```diff
@@ -0,0 +1,342 @@
+#
```
Comment (dirkmueller) [inline mkcloudruns/hacks/conduitresolver.rb]: this should rather be added to the crowbar / crowbar-core repository and installed + made available for any customer. no need to keep that in a hacks/ subdirectory.

**Comment by dirkmueller** [comment]
Personally I'd love to see the work on having a python wrapper that reads two oslo.config config files being revived (one for the host specific setup parameters and one for the scenario parameters).

**Comment by bmwiedemann** [comment]
@dguitarbite did you have a look at the codeclimate results? While some of them might be ignored (e.g. a function with 32 lines being 2 too many), others might point to things worth improving on.

**Comment by dguitarbite** [comment]
@bmwiedemann Yes, I took a look at the code climate and additionally I am waiting for people to give me feedback. I may be a bit slow on updating this PR at present. But please point out the issues :+1: . As I mentioned earlier, I am pushing everything I have and there was not enough time and feedback for all the things in here.

@dirkmueller Yes, let me think about this and do the same.

**Comment by aspiers** [comment]
@bmwiedemann I'm not sure why this was merged with a `wip` label? There are still a lot of issues and it's not very usable currently.

**Comment by aspiers** [comment]
No point having all these people assigned now that it's merged.

**Comment by dguitarbite** [comment]
Adam, I agree with you. This feeling is repeating. Although the current
code review system is much nicer, it still leaves open room for merging
PR's (2x+1's) while existing -1's or comments unaddressed are present.

**Comment by aspiers** [comment]
I'd be half-inclined to revert this PR if it wasn't for the fact that it submitted a newer cleaner version than what's in the `mkcloudruns` repo ...

**Comment by dguitarbite** [comment]
Adam, I agree with you. As promised (if not I promise) to add all the
fixes. I simply ran out of time.

**Pull request merged**